code/game/machinery/pipe/construction.dm

```diff
@@ -165,6 +165,11 @@ Buildable meters
 
 	if(pipename)
 		A.name = pipename
+	if("on" in A.vars)

```

Comment:
just move the var up, there's only 3 out of 16 objects in /component that don't use a `on` var anyway.